examples/dao/setup_sigma_dao.sql

```diff
@@ -60,31 +70,74 @@ BEGIN
 END;
 $$ LANGUAGE plpgsql;
 
+-- create a procedure to handle the trigger (sigmadao_proposals_trigger_fn) action
+CREATE OR REPLACE FUNCTION sigmadao_proposals_trigger_fn()
+RETURNS TRIGGER
+AS $$
+DECLARE
+	voting_start_key CHAR(255) := 'dm90aW5nX3N0YXJ0'; -- Byte code of 'voting_end'
+	voting_end_key CHAR(255) := 'dm90aW5nX2VuZA=='; -- Byte code of 'voting_end'
+	voting_start_value BIGINT;
+	voting_end_value BIGINT;
+BEGIN
+	IF (SELECT NEW.localstate::jsonb -> 'tkv' -> voting_start_key) IS NOT NULL THEN
+		-- Iterate json object and get voting start value
+		SELECT NEW.localstate::jsonb -> 'tkv' -> voting_start_key -> 'ui' INTO voting_start_value;
+		-- Iterate json object and get voting end value
+		SELECT NEW.localstate::jsonb -> 'tkv' -> voting_end_key -> 'ui' INTO voting_end_value;
+		-- insert values in sigma_dao_proposals table
+		INSERT INTO public.sigma_dao_proposals (
+		addr, app, localstate, voting_start, voting_end)
+		VALUES (NEW.addr, NEW.app, NEW.localstate::jsonb, voting_start_value, voting_end_value);
+	END IF;
+	RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
 -- create a trigger (sigmadao_trigger)
 -- Event: INSERT on publoc.txn relation
 CREATE TRIGGER sigmadao_trigger
 AFTER INSERT ON public.txn FOR EACH ROW
 EXECUTE FUNCTION sigmadao_trigger_fn();
 
--- grant privileges to relation sigma_daos
-GRANT ALL PRIVILEGES ON TABLE sigma_daos TO algorand;
+-- create a trigger (sigmadao_proposals_trigger)
+-- Event: UPDATE on publoc.account_app relation
+CREATE TRIGGER sigmadao_proposals_trigger
+AFTER UPDATE ON public.account_app FOR EACH ROW
+EXECUTE FUNCTION sigmadao_proposals_trigger_fn();
+
+-- grant privileges
+GRANT ALL PRIVILEGES ON TABLE sigma_daos, sigma_dao_proposals TO algorand;
 
 -- Below are objects needed for sigma dao app. Extend it if more sigma dao objects needed
 -- Here, sigma_dao_user should be same as https://github.com/scale-it/algo-builder/blob/cbc2123622a10fbf96f9b99b254abc86a79ac1fb/examples/dao/Makefile#L1
-GRANT SELECT ON TABLE sigma_daos, asset, account_asset, account_app TO sigma_dao_user;
+GRANT SELECT ON TABLE sigma_daos, sigma_dao_proposals, asset, account_asset, account_app TO sigma_dao_user;
 
 -- Function to search dao name in sigma_daos relation
 CREATE OR REPLACE FUNCTION search_sigma_daos(daoToBeSearched TEXT)
 RETURNS SETOF sigma_daos AS $$
 	SELECT * FROM sigma_daos WHERE dao_name ilike ('%' || daoToBeSearched || '%');
 $$ LANGUAGE SQL STABLE;
 
--- Function to search proposal in account_app relation by app id
-CREATE OR REPLACE FUNCTION search_sigma_daos_proposals(appId BIGINT)
-RETURNS SETOF account_app AS $$
+-- Function to search proposal in sigma_dao_proposals relation by app id
+CREATE OR REPLACE FUNCTION sigma_daos_proposal_filter(appId BIGINT, filterType INT)
```

Comment:
do we need to add a pagination info? Maybe we can add it later.

Reply:
No, we don't need that. Postgraphile handles that for us.

Reply:
but this functions returns all records. We don't want to extract all records from DB if it's not necessary.

Reply:
Yes, we need all the details.

Reply:
record = row. This function doesn't know if it should return only first 100 proposals or the 100-199 proposals.